pkg/build/strategies/sti/sti.go

```diff
@@ -216,22 +216,29 @@ func (b *STI) PostExecute(containerID string, location string) error {
 
 	buildEnv := append(scripts.ConvertEnvironment(env), b.generateConfigEnv()...)
 
-	cmd := []string{}
 	opts := docker.CommitContainerOptions{
-		Command:     append(cmd, filepath.Join(location, api.Run)),
+		Command:     append([]string{}, filepath.Join(location, api.Run)),
 		Env:         buildEnv,
 		ContainerID: containerID,
-		Repository:  b.request.Tag,
 	}
+	if len(b.request.Tag) > 0 {
+		opts.Repository = b.request.Tag
+	}
+
```

Comment:
Do you really need this change? Wouldn't opts.Repository remain blank if you just assign it b.request.Tag (if blank)?